libsolidity/codegen/YulUtilFunctions.cpp

```diff
@@ -1419,23 +1427,27 @@ string YulUtilFunctions::storageArrayPushFunction(ArrayType const& _type)
 					default {
 						sstore(array, add(data, 2))
 						let slot, offset := <indexAccess>(array, oldLen)
-						<storeValue>(slot, offset, value)
+						<storeValue>(slot, offset, <values>)
 					}
 				<!isByteArray>
 					let oldLen := sload(array)
 					if iszero(lt(oldLen, <maxArrayLength>)) { <panic>() }
 					sstore(array, add(oldLen, 1))
 					let slot, offset := <indexAccess>(array, oldLen)
-					<storeValue>(slot, offset, value)
+					<storeValue>(slot, offset, <values>)
 				</isByteArray>
 			})")
 			("functionName", functionName)
+			("values", suffixedVariableNameList("value", 0, _fromType->sizeOnStack()))
 			("panic", panicFunction(PanicCode::ResourceError))
 			("extractByteArrayLength", _type.isByteArray() ? extractByteArrayLengthFunction() : "")
 			("dataAreaFunction", arrayDataAreaFunction(_type))
 			("isByteArray", _type.isByteArray())
 			("indexAccess", storageArrayIndexAccessFunction(_type))
-			("storeValue", updateStorageValueFunction(*_type.baseType(), *_type.baseType()))
+			(
+				"storeValue",
+				updateStorageValueFunction(_fromType ? *_fromType : *_type.baseType(), *_type.baseType())
```

Comment:
```suggestion
				updateStorageValueFunction(*_fromType)
```
or you can get rid of the ``if (!_fromType) _fromType = _type.baseType();`` above, can't you :-)?